lib/Data/Validate/Sanctions/Redis.pm

```diff
@@ -0,0 +1,230 @@
+package Data::Validate::Sanctions::Redis;
+
+use strict;
+use warnings;
+
+use parent 'Data::Validate::Sanctions';
+
+use Data::Validate::Sanctions::Fetcher;
+use Scalar::Util    qw(blessed);
+use List::Util      qw(max);
+use JSON::MaybeUTF8 qw(encode_json_utf8 decode_json_utf8);
+use Syntax::Keyword::Try;
+
+# VERSION
+
+sub new {
+    my ($class, %args) = @_;
+
+    my $self = {};
+    $self->{redis_read}  = $args{redis_read} or die 'Redis read connection is missing';
+    $self->{redis_write} = $args{redis_write};
+
+    $self->{sources} = [keys Data::Validate::Sanctions::Fetcher::config(eu_token => 'dummy')->%*];
+
+    $self->{args}      = {%args};
+    $self->{last_time} = 0;
+    my $object = bless $self, ref($class) || $class;
+    $object->_load_data();
+
+    return $object;
+}
+
+sub last_updated {
+    my $self = shift;
+    my $list = shift;
+
+    if ($list) {
+        return $self->{_data}->{$list}->{updated};
+    } else {
+        $self->_load_data();
+        return max(map { $_->{updated} } values %{$self->{_data}});
+    }
+}
+
+sub set_sanction_file {
+    die 'Not applicable';
+}
+
+sub get_sanction_file {
+    die 'Not applicable';
+}
+
+sub get_sanctioned_info {
+    my $self = shift;
+
+    die "This function can only be called on an object" unless $self;
+
+    return Data::Validate::Sanctions::get_sanctioned_info($self, @_);
+}
+
+sub _load_data {
+    my $self = shift;
+
+    $self->{last_time}               //= 0;
+    $self->{_data}                   //= {};
+    $self->{_sanctioned_name_tokens} //= {};
+    $self->{_token_sanctioned_names} //= {};
+
+    my $last_time = $self->{last_time};
+    for my $source ($self->{sources}->@*) {
+        try {
+            my $updated = $self->{redis_read}->hget("SANCTIONS::$source", 'updated') // 0;
+            next if $updated <= ($self->{_data}->{$source}->{updated} // 0);
+
+            $self->{_data}->{$source}->{content}  = decode_json_utf8($self->{redis_read}->hget("SANCTIONS::$source", 'content'));
+            $self->{_data}->{$source}->{verified} = $self->{redis_read}->hget("SANCTIONS::$source", 'verified');
```

Comment:
How about using hmget to get both keys at once?

Reply:
Done. Thanks.